### PR TITLE
Ramda: Tighten type signature for `append`

### PR DESCRIPTION
**Pull request by MazeChaZer**
[The Ramda documentation clearly states](http://ramdajs.com/docs/#append) that the element to append an the elements in the list should have the same type.
The previous type definition unnecessarily weakened type safety. For example this error could not be detected by TypeScript:

```typescript
interface A {
    id: number
}
const list: A[] = []
R.append({idd: 2}, list)
```

Which seems ridiculous. If one wanted an intersection type, one could explicity define it to be used for the list/element type. Additionally, the function `prepend` already is as strict as this proposal, that should probably be consistent.

I also deleted a redundant line.

### CODE REVIEW

**Comment by MazeChaZer** [comment]
This issue is already solved by #18763

**Comment by MazeChaZer** [comment]
Reopening, because #18763 was closed and #20177 was abandoned.
Can any of the authors (@donnut @mdekrey @LiamGoodacre @mrdziuban @sbking @afharo @teves-castro @1m0rebug @hojberg @charlespwd) review this 9 line change? The previous broken type annotation for `append` caused multiple runtime errors and annoying bugs for us.

**Review by charlespwd — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
